lib/tasks/refresh_prisoner_category.rake

```diff
@@ -6,7 +6,12 @@ namespace :prisoner_category do
 
     puts "Refreshing #{count} profiles"
     profiles.find_each.with_index do |profile, i|
-      Profiles::ImportPrisonerCategory.new(profile).call
+      latest_nomis_booking_id = profile&.person&.latest_nomis_booking_id
+      if latest_nomis_booking_id.present?
+        profile.category = Categories::FindByNomisBookingId.new(latest_nomis_booking_id).call
```

Comment:
I think assigning a category this way actually updates the record?

Reply:
😱 arghhhhh 😱

Reply:
@cwrw I've double-checked and the assignment _does not_ save the record - so I think this is fine:

```
[39] pry(main)> p.category
=> nil
[40] pry(main)> p.category = Category.first
  Category Load (0.4ms)  SELECT "categories".* FROM "categories" ORDER BY "categories"."id" ASC LIMIT $1  [["LIMIT", 1]]
=> #<Category:0x00007f97c881a9e0 id: "1759804b-f958-44f9-b085-fd99bf7fbdbd", key: "CYOI", title: "Category C / YOI", move_supported: true, created_at: Tue, 03 Nov 2020 11:17:58 GMT +00:00, updated_at: Tue, 03 Nov 2020 11:17:58 GMT +00:00>
[41] pry(main)> p.updated_at
=> Fri, 06 Nov 2020 16:00:45 GMT +00:00
[42] pry(main)> p.save
   (0.3ms)  BEGIN
  Profile Update (0.7ms)  UPDATE "profiles" SET "updated_at" = $1, "category_id" = $2 WHERE "profiles"."id" = $3  [["updated_at", "2020-11-06 16:01:13.895585"], ["category_id", "1759804b-f958-44f9-b085-fd99bf7fbdbd"], ["id", "6388e8bf-9845-466a-8540-8f31d800fea1"]]
  PaperTrail::Version Create (0.4ms)  INSERT INTO "versions" ("item_type", "item_id", "event", "object", "created_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["item_type", "Profile"], ["item_id", "6388e8bf-9845-466a-8540-8f31d800fea1"], ["event", "update"], ["object", "---\nid: 6388e8bf-9845-466a-8540-8f31d800fea1\nperson_id: 0003a13f-a6e4-4b33-aa57-60c172f993e7\ncreated_at: &1 2020-11-06 16:00:45.836746000 Z\nupdated_at: *1\nassessment_answers: \"[]\"\ncategory_id: \n"], ["created_at", "2020-11-06 16:01:13.895585"]]
  Person Update (0.4ms)  UPDATE "people" SET "updated_at" = $1 WHERE "people"."id" = $2  [["updated_at", "2020-11-06 16:01:13.899660"], ["id", "0003a13f-a6e4-4b33-aa57-60c172f993e7"]]
  PaperTrail::Version Create (0.3ms)  INSERT INTO "versions" ("item_type", "item_id", "event", "object", "created_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["item_type", "Person"], ["item_id", "0003a13f-a6e4-4b33-aa57-60c172f993e7"], ["event", "update"], ["object", "---\nid: 0003a13f-a6e4-4b33-aa57-60c172f993e7\ncreated_at: 2020-07-09 17:49:53.898335000 Z\nupdated_at: 2020-11-06 16:01:13.899660000 Z\nnomis_prison_number: \nprison_number: \ncriminal_records_office: \npolice_national_computer: 0TFIZ1/VZ\nfirst_names: Jett\nlast_name: Price\ndate_of_birth: 1955-06-02\ngender_additional_information: ''\nethnicity_id: 379ced2e-a3f2-44dd-8ebe-e935e600ad68\ngender_id: 1b227566-ef41-4516-b41d-9aec6651243a\nlast_synced_with_nomis: \nlatest_nomis_booking_id: \n"], ["created_at", "2020-11-06 16:01:13.899660"]]
   (5.7ms)  COMMIT
=> true
[43] pry(main)> p.updated_at
=> Fri, 06 Nov 2020 16:01:13 GMT +00:00
```

Reply:
ah your right, its only on the has many, single ones don't get updated, phew! It caught me out a bit when updating flags but this is fine 👍